### ISSUE TITLE
Follow menu falls under post links

### ISSUE THREAD

**Issue by KiarashS** [mmistakes/minimal-mistakes#2478]
I use latest version of `minimal-mistakes`, and when I tap `Follow` button in mobile devices, `Follow menu` falls under post links.

![follow](https://user-images.githubusercontent.com/1054134/78759188-7d263c80-7994-11ea-8ffb-1987ab585e0e.JPG)

**Comment by iBug** [comment]
I managed to reproduce this on <https://blog.kiarash.pro/>. But to be honest, your setup looks strange to me in that instead of linking to the actual post, the title links to somewhere else. I'm afraid it's not the intended usage of the "article menu" page.

**Comment by KiarashS** [comment]
@iBug  That's OK, because `Post link` is a feature of `minimal-mistake`.
According to [here](https://github.com/mmistakes/minimal-mistakes/blob/master/test/_posts/2010-03-07-post-link.md).

**Comment by mmistakes** [comment]
Looks like a z-indexing issue when I refactored the follow me menu styling a few releases back. I don't personally use link posts so didn't notice the overlap.

Those elements have `z-index: 10` which puts them over the follow links. [Changing it to something like `5`](https://github.com/mmistakes/minimal-mistakes/blob/master/_sass/minimal-mistakes/_archive.scss#L26) seems to fix it.

```scss
.archive__item a {
    position: relative;
    z-index: 5;
}
```

If someone wants to submit a PR to fix that would be great. No idea when I'll get to it.

![image](https://user-images.githubusercontent.com/1376749/78787702-c072b200-7978-11ea-9169-3c4a524f34f8.png)

**Comment by mmistakes** [comment]
Looks like @iBug already took care of the fix in #2478. Thanks.
